microovn/ovn/join.go

```diff
@@ -140,13 +142,60 @@ func Join(s *state.State, initConfig map[string]string) error {
 		return fmt.Errorf("Failed to get OVN SB connect string: %w", err)
 	}
 
+	// Encapsulation IP configuration
+
+	// A custom encapsulation IP address can be defined for a join member by a bootstrapping member.
+	// In this case, the custom IP address will be stored in the database by the bootstrapping member
+	// after its creation (see MicroOVM init CLI usage).
+	var ovnEncapIp string
+	err = s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		item, err := database.GetConfigItem(ctx, tx, fmt.Sprintf("%s.ovn-encap-ip", s.Name()))
+		if err != nil {
+			if err == api.StatusErrorf(http.StatusNotFound, "ConfigItem not found") {
+				return nil
+			}
+
+			return err
+		}
+
+		ovnEncapIp = item.Value
+
+		// Remove the config item as we only need it one time for the OVS configuration.
+		err = database.DeleteConfigItem(ctx, tx, item.Key)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("Failed to get custom encapsulation IP address from database: %w", err)
+	}
+
+	// A custom encapsulation IP address can also be directly passed as an initConfig parameter.
+	// This is typically the case when an other component relying on MicroOVN (e.g, MicroCloud)
```

Comment:
nit: These comments are very helpfull but tather than mentioning downstream projects, I think it would be better to make it clearer that the block above (pulling the encapsulation IP from config DB) is taken when user does plain `microovn cluster join` and below is the more configurable way when user uses `microovn cluster init`